spring-integration-redis/src/test/java/org/springframework/integration/redis/rules/RedisAvailableTests.java

```diff
@@ -85,6 +92,26 @@ protected void awaitContainerSubscribedWithPatterns(RedisMessageListenerContaine
 		Thread.sleep(1000);
 	}
 
+	protected void awaitFullySubscribed(RedisMessageListenerContainer container, RedisTemplate<?, ?> redisTemplate,
+			String redisChannelName, QueueChannel channel, Object message) throws Exception {
+		awaitContainerSubscribedNoWait(container);
+		drain(channel);
+		long now = System.currentTimeMillis();
+		Message<?> received = null;
+		while (received == null && System.currentTimeMillis() - now < 10000) {
+			redisTemplate.convertAndSend(redisChannelName, message);
+			received = channel.receive(1000);
+		}
+		drain(channel);
+		assertNotNull("Container failed to fully start", received);
+	}
+
+	private void drain(QueueChannel channel) {
+		while (channel.receive(0) != null) {
+			;
+		}
```

Comment:
Isn't `QueueChannel.clear()` enough for us? 
Actually it does the same:
```
while ((message = this.queue.poll()) != null) {
	clearedMessages.add(message);
}
```
Where `channel.receive(0)` is equivalent to:
```
if (timeout == 0) {
	return this.queue.poll();
}
```
So, will make code more cleaner on merge if I agree with all other changes